### PR TITLE
Feaute/custom text view

### PR DESCRIPTION
**Pull request by leidyzulu**
Show Textview type Title or Subtitle calling it from the xml

### CODE REVIEW

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/CustomTextViewTest.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    @SmallTest
+    @Test
+    fun shouldShowTextTypeTitle() {
+        MockActivity.layout = R.layout.activity_custom_text_view_title
+        restartActivity()
+        // Given
+
+        // When
+
+        // Then
```
Comment (otigreros) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/CustomTextViewTest.kt]: why the empty given, when then scopes?

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/CustomTextViewTest.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    @SmallTest
+    @Test
+    fun shouldShowTextTypeSubtitle() {
+        MockActivity.layout = R.layout.activity_custom_text_view_subtitle
+        restartActivity()
+        // Given
+
+        // When
```
Comment (otigreros) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/CustomTextViewTest.kt]: +1

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/CustomTextViewTest.kt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    @SmallTest
+    @Test
+    fun shouldShowTextTypeDefault() {
+        MockActivity.layout = R.layout.activity_custom_text_view_default_text
+        restartActivity()
+        // Given
+
+        // When
```
Comment (otigreros) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/CustomTextViewTest.kt]: +1

File: app/src/main/res/values/attrs.xml [hunk 4]
```diff
@@ -63,4 +63,8 @@
         <attr name="type" format="string"/>
     </declare-styleable>
     <attr name="textInputLayoutTheme" format="reference"/>
+
+    <declare-styleable name="CustomTextView">
```
Comment (EAlexisDuque) [inline app/src/main/res/values/attrs.xml]: 1. Change this to an array with the *title* and *subtitle* options.

File: app/src/main/java/co/condorlabs/customcomponents/customtextview/CustomTextView.kt [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+import android.content.Context
+import android.graphics.Typeface
+import android.util.AttributeSet
+import android.util.TypedValue
+import androidx.appcompat.widget.AppCompatTextView
+import androidx.core.content.ContextCompat
+import co.condorlabs.customcomponents.*
+
+class CustomTextView(context: Context, private val attrs: AttributeSet) : AppCompatTextView(context, attrs) {
```
[the remark this answers is not in the document]
Comment (leidyzulu) [reply app/src/main/java/co/condorlabs/customcomponents/customtextview/CustomTextView.kt]: Show Textview type Title or Subtitle calling it from the xml

File: app/src/main/res/values/attrs.xml (hunk 4 above)
Comment (leidyzulu) [reply app/src/main/res/values/attrs.xml]: done

**Comment by EAlexisDuque** [comment]
1. Rename the methods of the *CustomTextView class* so they are more verbose.
2. Implement test for change its properties programatically.

File: app/src/main/java/co/condorlabs/customcomponents/customtextview/CustomTextView.kt [hunk 6]
```diff
@@ -4,52 +4,40 @@ import android.content.Context
...
-import android.view.View
 import androidx.appcompat.widget.AppCompatTextView
 import androidx.core.content.ContextCompat
 import co.condorlabs.customcomponents.*
 
 class CustomTextView(context: Context, private val attrs: AttributeSet) : AppCompatTextView(context, attrs) {
 
-    private var typeFont: Int = NO_ID
+    private var typeText: Int = ZERO
+    private var customTextViewStyleFactory = CustomTextViewStyleFactory()
```
Comment (EAlexisDuque) [inline app/src/main/java/co/condorlabs/customcomponents/customtextview/CustomTextView.kt]: You do not need to save *this variable*, you should use a `kotlin object`.

File: app/src/main/java/co/condorlabs/customcomponents/customtextview/CustomTextViewStyleFactory.kt [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+import co.condorlabs.customcomponents.TYPE_BODY
+import co.condorlabs.customcomponents.TYPE_SUBTITLE
+import co.condorlabs.customcomponents.TYPE_TITLE
+
+class CustomTextViewStyleFactory {
+
+    private val styles = SparseArray<CustomTextViewStyle>()
+
+    fun getStyleFromType(type: Int): CustomTextViewStyle {
+        styles.put(TYPE_BODY, CustomTextViewStyle.BodyTextViewStyle)
```
Comment (EAlexisDuque) [inline app/src/main/java/co/condorlabs/customcomponents/customtextview/CustomTextViewStyleFactory.kt]: Was `default` the name of the constant?
Why body?

**Comment by EAlexisDuque** [comment]
Update `readme` file.

File: app/src/main/java/co/condorlabs/customcomponents/customtextview/CustomTextViewStyleFactory.kt [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+        styles.put(TYPE_TITLE, CustomTextViewStyle.TitleTextViewStyle)
+        styles.put(TYPE_SUBTITLE, CustomTextViewStyle.SubTitleTextViewStyle)
+        return styles[type] ?: throw StyleNotFoundForType(type)
+    }
+}
```
[the remark this answers is not in the document]
Comment (leidyzulu) [reply app/src/main/java/co/condorlabs/customcomponents/customtextview/CustomTextViewStyleFactory.kt]: Sorry my fault

File: app/src/main/java/co/condorlabs/customcomponents/Constants.kt [hunk 9]
```diff
@@ -188,5 +188,6 @@ const val PROGRESS_DRAWABLE_SIZE_MULTIPLIER = 2
 const val PROGRESS_DRAWABLE_DIAMETER_DIVIDER = 2
 const val PROGRESS_DRAWABLE_LAYER_INSET_INDEX = 1
 
-const val TEXT_VIEW_TITLE = 0
-const val TEXT_VIEW_SUBTITLE = 1
+const val TYPE_TITLE = 1
```
[the remark this answers is not in the document]
Comment (leidyzulu) [reply app/src/main/java/co/condorlabs/customcomponents/Constants.kt]: Oscar says all constants should be in Constants.kt

File: test/src/main/res/layout/activity_baseedittextfield_multiline_test.xml [hunk 10]
```diff
@@ -13,7 +13,7 @@
             app:min_lines="5"
             app:max_lines="8"
             app:multiline="true"
-            android:textSize="@dimen/default_text_size"
+            android:textSize="@dimen/body"
             app:input_type="multiline"/>
 
 </LinearLayout>
```
Comment (otigreros) [inline test/src/main/res/layout/activity_baseedittextfield_multiline_test.xml]: blank line

**Pull request merged**